line_parser.c

```diff
@@ -40,14 +42,15 @@ DataInstruction* parse_data_instruction(char *line) {
                 operand_1 = NULL;
             }
             instruction->operand_1 = trim_whitespace(operand_1);
-
+            /* Maybe I'm missing something here, but it seems wrong. When there's only one operand it should be considered as operand_2 (the destination operand) */
             strncpy(operand_2, token + i + 1, strlen(token) - i);
             if (strlen(operand_2) == 0) {
                 printf("Invalid empty operand\n");
                 exit(1);
             }
 
             instruction->operand_2 = trim_whitespace(operand_2);
+            /* Also here - before returning the instruction - make sure this instruction should indeed have the number of arguments it got */
```

Comment:
it doesnt really matter where we do the validation

Reply:
You're right, since the code needs to run all the way down anyway